### PR TITLE
SPORTS-71 Add APIs to add/edit/get/delete fixtures.

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
- Add new APIs
  ```
  POST	/chess-fixtures
  GET	    /chess-fixtures/:id
  PUT	    /chess-fixtures/:id
  DELETE	/chess-fixtures/:id
  ```
- Add new Errors
  ```
  BadRequestError     400 Bad Request
  AlreadyExistsError  409 Conflict
  ```

_**NOTE:** Run `yarn migrate` after merge._

### CODE REVIEW

File: backend/src/errors/BadRequestError.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  /**
+   * Error message to be thrown.
+   *
+   * @type {string}
+   * @memberof ForbiddenError
+   */
+  message: string;
+
+  /**
+   * Creates an instance of ForbiddenError.
```
Comment (arun-tamang) [inline backend/src/errors/BadRequestError.ts]: instance of BadRequestError.

File: backend/src/errors/AlreadyExistsError.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  /**
+   * Error message to be thrown.
+   *
+   * @type {string}
+   * @memberof ForbiddenError
+   */
+  message: string;
+
+  /**
+   * Creates an instance of ForbiddenError.
```
Comment (arun-tamang) [inline backend/src/errors/AlreadyExistsError.ts]: instance of AlreadyExistsError

File: backend/src/services/chessFixtureService.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+      });
+    }
+
+    const chessFixture: ChessFixture = await chessFixtureQuery.fetch();
+
+    if (chessFixture) {
+      return true;
+    }
+
+    const chessFixture2Query: ChessFixture = await new ChessFixture().where({
```
Comment (arun-tamang) [inline backend/src/services/chessFixtureService.ts]: Isn't this the same as chessFixtureQuery? Argument passed in where is the same. And the subsequent ignoreIds.forEach is also doing the same thing with both of them.

File: backend/src/errors/AlreadyExistsError.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  message: string;
+
+  /**
+   * Creates an instance of AlreadyExistsError.
+   *
+   * @param {string} message
+   * @memberof ForbiddenError
+   */
+  constructor(message: string) {
+    super(message, HttpStatus.BAD_REQUEST);
```
Comment (arun-tamang) [inline backend/src/errors/AlreadyExistsError.ts]: How about another 409 code?
[https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists](url)

File: backend/src/constants/messages.ts [hunk 5]
```diff
@@ -101,3 +101,38 @@ export const teamPlayerMessages = {
   unableToUpdate: 'Unable to update the team player.',
   unableToCreate: 'Unable to create a new team player.'
 };
+
+export const chessFixtureMessages = {
```
Comment (aviskarkc10) [inline backend/src/constants/messages.ts]: make all of these messages `fixtures` instead of a single related to only chess

Comment (aviskarkc10) [reply backend/src/constants/messages.ts]: I see `fixtureStatusMessages` is already there

File: backend/src/services/chessFixtureService.ts [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+  try {
+    // Check existence of the round
+    const round: Round = await new Round({ id: params.round_id }).fetch();
+
+    if (!round) {
+      throw new NotFoundError(roundMessages.notFound);
+    }
+
+    // Check existence of the first team
+    const team1: Team = await new Team({ id: params.team_1_id }).fetch();
```
[the remark this answers is not in the document]
Comment (aviskarkc10) [reply backend/src/services/chessFixtureService.ts]: Players are always assigned a team. I think that's why @sanjeevkpandit has used team

File: backend/src/services/chessFixtureService.ts [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+ * @throws {NoRowUpdatedError|error}
+ */
+export async function remove(id: number) {
+  try {
+    const chessFixture: ChessFixture = await findById(id);
+    const removedChessFixture: ChessFixture = await chessFixture.destroy();
+
+    if (!removedChessFixture) {
+      throw new NoRowUpdatedError(chessFixtureMessages.unableToRemove);
+
```
Comment (aviskarkc10) [inline backend/src/services/chessFixtureService.ts]: unnecessary extra line here

**Pull request merged**